src/snmalloc/backend_helpers/buddy.h

```diff
@@ -70,8 +82,11 @@ namespace snmalloc
           size *= 2;
           addr = Rep::align_down(addr, size);
           if (size == bits::one_at_bit(MAX_SIZE_BITS))
+          {
+            invariant();
```

Comment:
Sorry if this should be obvious, but... Does this actually test anything?  `empty_above` should be `idx + 1` and `trees.size()` should be `MAX_SIZE_BITS - MIN_SIZE_BITS`, which should be equal to `idx + 1`?

Reply:
I wanted to ensure the invariant held on all paths out of the `add_block`, this was a sufficient set to achieve that.  So I placed it before each non-tail call return without thinking about what the invariant was.

I think you are right it trivially holds here.  However, I would like to leave it as the shape is nice.  But I'll add a comment.